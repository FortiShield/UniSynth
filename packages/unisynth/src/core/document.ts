/** Contains additional items which Unisynth patches on the Document. */
export interface UnisynthDocument extends Document {
  __q_context__?: [HTMLElement, Event, URL] | {};
}

export const _is_module = true;
