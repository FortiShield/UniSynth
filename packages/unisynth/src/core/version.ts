/**
 * UNISYNTH_VERSION
 *
 * @public
 */
export const version = (globalThis as any).UNISYNTH_VERSION as string;
