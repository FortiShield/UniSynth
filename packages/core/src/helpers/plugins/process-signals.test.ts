import { types } from '@babel/core';
import { capitalize } from '../capitalize';
import { replaceSignalSetters } from './process-signals';

describe(replaceSignalSetters.name, () => {
  test('should replace signal setters', () => {
    const code = `
    props.khulnasoftContextSignal.value.content = {
      ...khulnasoftContextSignal.content,
      ...newContent,
      data: {     
        ...khulnasoftContextSignal.content?.data, 
        ...newContent?.data },
        meta: {
          ...khulnasoftContextSignal.content?.meta,
          ...newContent?.meta,
          breakpoints:
          newContent?.meta?.breakpoints ||
          khulnasoftContextSignal.content?.meta?.breakpoints,
        }
        };
        
    khulnasoftContextSignal.value.rootState = newRootState;
    `;
    const propName = 'khulnasoftContextSignal';

    const output = replaceSignalSetters({
      code,
      nodeMaps: [
        {
          from: types.memberExpression(
            types.memberExpression(types.identifier('props'), types.identifier(propName)),
            types.identifier('value'),
          ),
          setTo: types.memberExpression(
            types.identifier('props'),
            types.identifier('set' + capitalize(propName)),
          ),
        },
        {
          from: types.memberExpression(types.identifier(propName), types.identifier('value')),
          setTo: types.identifier('set' + capitalize(propName)),
        },
      ],
    });
    expect(output).toMatchSnapshot();
  });
});
