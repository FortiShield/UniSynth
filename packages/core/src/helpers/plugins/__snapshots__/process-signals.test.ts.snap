// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`replaceSignalSetters > should replace signal setters 1`] = `
"props.setKhulnasoftContextSignal(PREVIOUS_VALUE => ({ ...PREVIOUS_VALUE,
  content: { ...khulnasoftContextSignal.content,
    ...newContent,
    data: { ...khulnasoftContextSignal.content?.data,
      ...newContent?.data
    },
    meta: { ...khulnasoftContextSignal.content?.meta,
      ...newContent?.meta,
      breakpoints: newContent?.meta?.breakpoints || khulnasoftContextSignal.content?.meta?.breakpoints
    }
  }
}));
setKhulnasoftContextSignal(PREVIOUS_VALUE => ({ ...PREVIOUS_VALUE,
  rootState: newRootState
}))"
`;
